_rules/element-lang-matches-common-language-off6ek.md

```diff
@@ -0,0 +1,339 @@
+---
+id: off6ek
+name: HTML element language subtag matches language
+rule_type: atomic
+description: |
+  This rule checks that the primary language subtag of an element matches its language
+accessibility_requirements:
+  wcag20:3.1.2: # Language of Parts (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+  wcag-technique:H58: # Using language attributes to identify changes in the human language
+    forConformance: false
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: satisfied
+input_aspects:
+  - DOM tree
+  - Accessibility tree
+  - CSS Styling
+  - Language
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # using aria-labelledby instead
+  - 'alt-require'
+---
+
+## Applicability
+
+This rule applies to any HTML element for which all the following are true:
+
+- the element is an [inclusive descendant][] in the [flat tree][] of a `body` element; and
+- The element is in a [top-level browsing context][]; and
+- The element is in a [document][] with a [content type][] of `text/html`; and
+- The element has a `lang` [attribute value][] which is a [valid language tag][]; and
+- The element has some non-empty [text with the same programmatic language][].
+
+## Expectation
+
+For each test target, the [primary language][] of the [valid language tag][] is a [common language][] of the test target.
```

Comment:
It took me an hour reviewing this rule before I figured out that you want (almost) every word to match the `lang` attribute. I don't think that's as clear as it should be. But more importantly, I think this is far too strict. This SC is applicable to passages and phrases, not to individual words.

Putting a lang attribute on every word isn't just burdensome to content authors, it's distracting to screen reader users. Imagine having an English sentence with 2 French words in it. That results in 5 changes in voice, English -> French -> English -> French -> English, all in the span of a few seconds. That's not the intent of SC 3.1.2.

I think we may need to do something different in this rule. Maybe we need to figure out how to objectively define phrase and passage, so that we can make the rule applicable to those?

Reply:
The discussion seems to be raging on already in https://github.com/w3c/wcag/issues/1174 essentially with similar arguments, I'd say…

* Against the "single word" interpretation:
    * WCAG says "passage or phrase", not "word, passage or phrase".
    * It's disturbing for screen-reader users, so it's actually decreasing the accessibility.
* For the "single word" interpretation:
    * The Understanding document has a paragraph (4th of Intent) explaining when single words need to be considered, why if single words shouldn't be considered?
    * In many cases, single words need to be pronounced differently to be correctly understood.
    * Single words that do not really need language markup are falling under the "vernacular language" exception, so it doesn't mean "every single word" but leaves room for interpretation.
    
Still no strong opinion on that on my side 🤷